packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -916,6 +918,50 @@ export const validateAndParseAddress = (
   return _validateAndParseAddressFn(address);
 };
 
+export async function getEstimatedFees(
+  network: Network,
+  address: string,
+  privateKey: string,
+  accountPublicKey: string,
+  transactionInvocations: Invocations,
+  transactionVersion: '0x2' | '0x3',
+  includeDeployFee: boolean,
+): Promise<EstimateFeeResponse> {
+  let bulkTransactions = transactionInvocations;
+
```

Comment:
it looks like some arg, like accountPublicKey only begin use when it has deploy,

i would suggest move isAccountDeploy from the rpc to this function

as at the end the func return a result includeDeployFee